apsuite/optimization.py

```diff
@@ -75,92 +131,159 @@ def _create_swarm(self):
         # Initializing with zero velocity
         self._velocity = np.zeros((self._nswarm, self._ndim))
 
-    def _set_lim(self):
+    def _update_position(self):
+        """."""
+        r_indiv = self._coeff_indiv * np.random.rand()
+        r_coll = self._coeff_coll * np.random.rand()
+        # Inertial velocity
+        self._velocity = self._coeff_inertia * self._velocity
+        # Velocity dependent to distance from best individual position
+        self._velocity += r_indiv * (self._best_particle - self._position)
+        # Velocity dependent to distance from best global position
+        self._velocity += r_coll * (self._best_global - self._position)
+        # Update position and check boundary limits
+        self._position = self._position + self._velocity
+        self._check_lim()
+
+    def _check_lim(self):
+        """."""
         # If particle position exceeds the boundary, set the boundary value
         for i in range(self._upper_limits.size):
             over = self._position[:, i] > self._upper_limits[i]
             under = self._position[:, i] < self._lower_limits[i]
             self._position[over, i] = self._upper_limits[i]
             self._position[under, i] = self._lower_limits[i]
 
-    def initialization(self):
-        pass
+    def set_limits(self, upper=None, lower=None):
+        """."""
+        self._upper_limits = upper
+        self._lower_limits = lower
+        self.ndim = len(upper)
+        if not self.nswarm:
+            self.nswarm = int(10 + 2 * np.sqrt(self.ndim))
 
-    def _check_initialization(self):
-        if len(self._upper_limits) != len(self._lower_limits):
-            print('Warning: Upper and Lower Limits has different lengths')
+    def get_change(self):
+        pass
 
-        if self._ndim != len(self._upper_limits):
-            print('Warning: Dimension incompatible with limits!')
+    def set_change(self):
+        pass
```

Comment:
I usually implement an abstrat base method like this:

```python
def set_change(self):
    """DocString."""
    raise NotImplementedError
```
I think it is safer to raise an exception when the abstract method is invoked. Otherwise the class user may be using the base class improperly without knowing it (and eventual bugs might be lurking in...)

Reply:
I suppose `set_change` is an abstract method to be implemented in a derived class... is this so?

Reply:
Yes, correct. Thanks!